packages/serverpod/lib/src/database/extensions.dart

```diff
@@ -101,150 +86,275 @@ extension TableComparisons on TableDefinition {
     return null;
   }
 
-  /// Returns true if the table structure is identical to the [other]. Ignores
-  /// comparisons of Dart types and names.
-  bool like(TableDefinition other) {
-    // Columns
-    if (other.columns.length != columns.length) {
-      return false;
+  /// Compares this table definition with [other], returning a list of mismatches.
+  /// The comparison checks columns, indexes, foreign keys, and general table properties.
+  /// Returns an empty list if the tables are identical.
+  List<String> like(TableDefinition other) {
+    List<String> mismatches = [];
+
+    // Compare table name
+    if (other.name != name) {
+      mismatches
+          .add('Table name mismatch: Expected "$name", found "${other.name}".');
+    }
+
+    // Compare table space
+    if (other.tableSpace != tableSpace) {
+      mismatches.add(
+          'Tablespace mismatch: Expected "$tableSpace", found "${other.tableSpace}".');
     }
+
+    // Compare table schema
+    if (other.schema != schema) {
+      mismatches
+          .add('Schema mismatch: Expected "$schema", found "${other.schema}".');
+    }
+
+    // Columns
     for (var column in columns) {
       var otherColumn = other.findColumnNamed(column.name);
-      if (otherColumn == null || !column.like(otherColumn)) {
-        return false;
+      if (otherColumn == null) {
+        mismatches
+            .add('Column "${column.name}" is missing in the target schema.');
+      } else {
+        var columnMismatches = column.like(otherColumn);
+        if (columnMismatches.isNotEmpty) {
+          mismatches.add(
+              'Column "${column.name}" mismatch: ${columnMismatches.join(', ')}');
+        }
       }
     }
 
-    // Indexes
-    if (other.indexes.length != indexes.length) {
-      return false;
+    for (var otherColumn in other.columns) {
+      var column = findColumnNamed(otherColumn.name);
+      if (column == null) {
+        mismatches.add(
+            'Extra column "${otherColumn.name}" found in the target schema.');
+      }
```

Comment:
Suggestion: Remove this new condition.

This is a new scenario that you introduced with this PR right?

I don't think we should introduce this together with this PR. I don't think the warning is bad in itself but we should separate adding this from just adding more detailed warnings.

Reply:
removed